### PR TITLE
Use the WASM version of customize_symbol

### PR DESCRIPTION
**Pull request by jessepinho**
In #896, I pulled `customizeSymbol` into the WASM package so that we could use it in the transaction planner. That resulted in the `customizeSymbol` code being duplicated between TypeScript and Rust.

This PR deuplicates that code by A) getting rid of the TypeScript version, B) updating existing TypeScript code to use the one exported from WASM, and C) adding tests to the WASM version.

### CODE REVIEW

File: packages/wasm/crate/src/metadata.rs [hunk 1]
```diff
@@ -8,10 +8,13 @@ use crate::{error::WasmResult, utils};
 
 pub static UNBONDING_TOKEN_REGEX: &str = "^uunbonding_(?P<data>start_at_(?P<start>[0-9]+)_(?P<validator>penumbravalid1(?P<id>[a-zA-HJ-NP-Z0-9]+)))$";
 pub static DELEGATION_TOKEN_REGEX: &str =
-    "^udelegation_(?P<data>penumbravalid1[a-zA-HJ-NP-Z0-9]+)$";
+    "^udelegation_(?P<data>penumbravalid1(?P<id>[a-zA-HJ-NP-Z0-9]+))$";
```
Comment (jessepinho) [inline packages/wasm/crate/src/metadata.rs:11]: Discovered this bug in the delegation token regex, now that we're using the WASM version of `customize_symbol`.

File: packages/wasm/crate/src/metadata.rs [hunk 2]
```diff
@@ -61,3 +66,117 @@ fn shorten_id(captures: &regex::Captures) -> WasmResult<String> {
...
+            display: String::from(display_denom),
+            images: Vec::new(),
+            name: String::from(""),
+            penumbra_asset_id: None,
+            symbol: String::from(""),
+        }
+    }
+
+    #[test]
+    fn it_interpolates_display_denom() {
```
Comment (jessepinho) [inline packages/wasm/crate/src/metadata.rs:102]: This is a test for a test helper, which... might be overkill 😆 but I was taking it as an opportunity to practice my newly acquired Rust testing skills.

File: packages/wasm/crate/src/metadata.rs [hunk 3]
```diff
@@ -8,10 +8,13 @@ use crate::{error::WasmResult, utils};
...
 pub static UNBONDING_TOKEN_REGEX: &str = "^uunbonding_(?P<data>start_at_(?P<start>[0-9]+)_(?P<validator>penumbravalid1(?P<id>[a-zA-HJ-NP-Z0-9]+)))$";
 pub static DELEGATION_TOKEN_REGEX: &str =
-    "^udelegation_(?P<data>penumbravalid1[a-zA-HJ-NP-Z0-9]+)$";
+    "^udelegation_(?P<data>penumbravalid1(?P<id>[a-zA-HJ-NP-Z0-9]+))$";
 pub static SHORTENED_ID_LENGTH: usize = 8;
 
 #[wasm_bindgen]
+/// Given a binary-encoded `Metadata`, returns a new binary-encoded `Metadata`
+/// with the symbol customized if the token is one of several specific types
+/// that don't have built-in symbols.
```
Comment (grod220) [inline packages/wasm/crate/src/metadata.rs:17]: Think the comments go above the `#[wasm_bindgen]` attribute right?

File: packages/wasm/crate/src/metadata.rs [hunk 4]
```diff
@@ -61,3 +66,117 @@ fn shorten_id(captures: &regex::Captures) -> WasmResult<String> {
         .take(SHORTENED_ID_LENGTH)
         .collect())
 }
+
+#[cfg(test)]
+mod test_helpers {
```
Comment (grod220) [inline packages/wasm/crate/src/metadata.rs:71]: Amazing, tests for our Rust code! 🤩

Comment (grod220) [reply packages/wasm/crate/src/metadata.rs:71]: Could you move this to the tests folder?

File: packages/types/src/customize-symbol.ts
Comment (grod220) [inline packages/types/src/customize-symbol.ts:1]: 0Ccwb

**Review by grod220 — APPROVED** [review]

File: packages/wasm/crate/src/metadata.rs (hunk 3 above)
Comment (jessepinho) [reply packages/wasm/crate/src/metadata.rs:16]: Ah, fixed

File: packages/wasm/crate/src/metadata.rs (hunk 4 above)
Comment (jessepinho) [reply packages/wasm/crate/src/metadata.rs:71]: Talked with Gabe on Discord — per the [Rust book](https://doc.rust-lang.org/book/ch11-03-test-organization.html#the-tests-directory), we'll just use the `tests` directory for integration tests.

**Pull request merged**